packages/backend/src/core/DiscoveryWatcher.ts

```diff
@@ -66,6 +79,28 @@ export class DiscoveryWatcher {
     this.logger.info('Update finished', { blockNumber })
   }
 
+  async compareWithCommitted(
+    name: string,
+    discovered: AnalyzedData[],
+    overrides?: Record<string, DiscoveryContract>,
+  ) {
+    const committed = await this.configReader.readDiscoveryJson(name)
+    const discoveredAsCommitted = prepareDiscoveryFile(discovered)
+
+    const diff = diffDiscovery(
+      committed.contracts,
+      discoveredAsCommitted.contracts,
+      overrides ?? {},
+    )
+
+    if (diff.length > 0) {
+      const message = `Detected changes for ${name}\n\n${diff
+        .map(diffToString)
+        .join('\n')}`
```

Comment:
This entire thing should be called `diffToMessage` and `diffToString` shoule be an internal part of it.

Also discord has message size limits, maybe we should truncate if message is larger. Truncating can be done in a separate pr.